src/ORM/DataList.php

```diff
@@ -126,23 +126,23 @@ public function alterDataQuery($callback)
             }
 
             return $list;
-        } else {
-            $list = clone $this;
-            $list->inAlterDataQueryCall = true;
+        }
 
-            try {
-                $res = call_user_func($callback, $list->dataQuery, $list);
-                if ($res) {
-                    $list->dataQuery = $res;
-                }
-            } catch (Exception $e) {
-                $list->inAlterDataQueryCall = false;
-                throw $e;
-            }
+        $list = clone $this;
+        $list->inAlterDataQueryCall = true;
 
+        try {
+            $res = $callback($this->dataQuery, $list);
```

Comment:
I think the failures are because this should be `$list->dataQuery`

Reply:
Whoops!